### PR TITLE
Added cosginatures to AggregateTransactions, added several new transaction types

### PR DESCRIPTION
**Pull request by the-realtom**
## Added:
- AggregateTransaction cosignatures
- MosaicMetadata
- AccountMetadata
- AccountRestriction transaction types
- Validation of transfer transaction
- Added missing types to Aggregate and inner transactions
- Updated all device tests to use new wallet path
- Added functionality to allow cosigning transactions
- Merged darren's NEM2 get public key changes

### CODE REVIEW

File: core/src/trezor/messages/Capability.py [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-    U2F = 14
+EOS = 6
+Ethereum = 7
+Lisk = 8
+Monero = 9
+NEM = 10
+Ripple = 11
+Stellar = 12
+Tezos = 13
+U2F = 14
```
Comment (chrdevmar) [inline core/src/trezor/messages/Capability.py:16]: do you know what caused this change? when I run `make gen` on this branch, this file on my machine is updated to include the `if nto utils.BITCOIN_ONLY` code in it

File: core/src/trezor/messages/NEM2CosignatureSignedTx.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+if __debug__:
+    try:
+        from typing import Dict, List, Optional
+        from typing_extensions import Literal  # noqa: F401
+    except ImportError:
+        Dict, List, Optional = None, None, None  # type: ignore
+
+
+class NEM2CosignatureSignedTx(p.MessageType):
+    MESSAGE_WIRE_TYPE = 808
```
Comment (chrdevmar) [inline core/src/trezor/messages/NEM2CosignatureSignedTx.py:14]: Likewise with the previous question, this file doesn't include `MESSAGE_WIRE_TYPE` on my machine when I run `make gen`

**Review by chrdevmar — COMMENTED** [review]
I'm not sure if its something wrong with me but whenever I run `make gen` now it breaks as the protobuf messages dont seem to being compiled properly into the python classes

File: core/src/trezor/messages/NEM2CosignatureSignedTx.py (hunk 2 above)
Comment (the-realtom) [reply core/src/trezor/messages/NEM2CosignatureSignedTx.py:14]: Mine didn't either I had to add this line manually. I'm not sure if the gen is meant to add this line or not though, but it definately needs to be there.

**Pull request merged**